### PR TITLE
application-pkg.oeclass: bugfixes

### PR DESCRIPTION
**Pull request by kimrhh**
- Use the new default file extension for images
 - Remove crosstool-NG leftovers

### CODE REVIEW

**Comment by esben** [comment]
When did tar.gz become default extension for rootfs images?

classes/rootfs.oeclass:
```
# archive recipe flags
CLASS_FLAGS += "rootfs_archive_formats rootfs_archive_tar_ext"
DEFAULT_USE_rootfs_archive_formats = "tar"
DEFAULT_USE_rootfs_archive_tar_ext = "tar"
ARCHIVE_IMAGE_FORMATS = "${USE_rootfs_archive_formats}"
ARCHIVE_IMAGE_DIRNAME = ""
ARCHIVE_IMAGE_TAR_EXT = "${USE_rootfs_archive_tar_ext}"
```

**Comment by esben** [comment]
Would it make sense to use USE_rootfs_archive_tar_ext in application-pkg.oeclass ?

**Comment by kimrhh** [comment]
I just now noticed that flag when looking in distro conf for other reason, yes of course we should use it instead

File: classes/application-pkg.oeclass [hunk 1]
```diff
@@ -37,7 +34,7 @@ addtask imageqaprep before configure after stage
 do_imageqaprep[dirs] = "${WORKDIR}/image-qa"
 do_imageqaprep[cleandirs] = "${WORKDIR}/image-qa"
 fakeroot do_imageqaprep () {
-    tar xf ${TARGET_SYSROOT}/${USE_rootfs_name}.tar
+    tar xf ${TARGET_SYSROOT}/${USE_rootfs_name}.${USE_rootfs_archive_tar_ext}
```
Comment (esben) [inline classes/application-pkg.oeclass]: You need to add "rootfs_archive_tar_ext" to CLASS_DEPENDS.

**Pull request merged**